dashboard/src/shared/utils.ts

```diff
@@ -276,3 +286,36 @@ export function getGlobalNamespaceOrNamespace(
       return "unknown";
   }
 }
+
+export function convertGrpcAuthError(e: any): CustomError | any {
+  const msg = e?.metadata?.headersMap?.["grpc-message"].toString();
+  switch (e?.code) {
+    case grpc.Code.Unauthenticated:
+      return new UnauthorizedError(msg);
+    case grpc.Code.FailedPrecondition:
+    case grpc.Code.Internal:
+      //TODO(agamez): this code shouldn't be returned by the API, but it is
+      if (["credentials", "unauthorized"].some(p => msg?.toLowerCase()?.includes(p))) {
```

Comment:
A bit weak, but I guess it is the only thing we can do until all error codes from backend are harmonized.